Patch-Source: https://gitlab.gnome.org/GNOME/gnome-control-center/-/merge_requests/3157
---
From 2dae05959c3eba620d2ae1e7ac236d2c0e74d058 Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Pablo=20Correa=20G=C3=B3mez?= <ablocorrea@hotmail.com>
Date: Wed, 28 May 2025 12:37:33 +0200
Subject: [PATCH 1/2] keyboard: include generated resource header files in the
 panel dep
MIME-Version: 1.0
Content-Type: text/plain; charset=UTF-8
Content-Transfer-Encoding: 8bit

This should fix a rare race where the global shortcuts panel has its
sources compiled before the keyboard resources are generated. Per the
meson documentation sources in declare_dependency: "sources to add to
targets (or generated header files that should be built before sources
including them are built)". Without the sources, the dependency makes
sure that the global shortcuts will not be linked until the dependency
is ready, but some of its files might be compiled before the header is
ready.

Example error:
https://gitlab.alpinelinux.org/fossdd/aports/-/jobs/1868755

ninja: job failed: cc -Iglobal-shortcuts-provider/gnome-control-center-global-shortcuts-provider.p -Iglobal-shortcuts-provider -I../global-shortcuts-provider -I. -I.. -Ipanels/common -I../panels/common -Ipanels/keyboard -I../panels/keyboard -Isubprojects/libgxdp/src -I../subprojects/libgxdp/src -Ishell -I/usr/include/glib-2.0 -I/usr/lib/glib-2.0/include -I/usr/include/libmount -I/usr/include/blkid -I/usr/include/gsettings-desktop-schemas -I/usr/include/libadwaita-1 -I/usr/include/gtk-4.0 -I/usr/include/pango-1.0 -I/usr/include/harfbuzz -I/usr/include/gdk-pixbuf-2.0 -I/usr/include/webp -I/usr/include/cairo -I/usr/include/freetype2 -I/usr/include/libpng16 -I/usr/include/pixman-1 -I/usr/include/graphene-1.0 -I/usr/lib/graphene-1.0/include -I/usr/include/fribidi -I/usr/include/appstream -I/usr/include/gio-unix-2.0 -I/usr/include/libwacom-1.0 -I/usr/include/gudev-1.0 -I/usr/include/libevdev-1.0 -flto=auto -fdiagnostics-color=always -D_FILE_OFFSET_BITS=64 -Wall -Winvalid-pch -DHAVE_CONFIG_H -Wno-deprecated-declarations -Os -fstack-clash-protection -Wformat -Werror=format-security -fno-plt -g -fPIE -mfpmath=sse -msse -msse2 -pthread -mfpmath=sse -msse -msse2 -mfpmath=sse -msse -msse2 -mfpmath=sse -msse -msse2 '-DGNOMELOCALEDIR="/usr/share/locale"' -MD -MQ global-shortcuts-provider/gnome-control-center-global-shortcuts-provider.p/cc-global-shortcut-dialog.c.o -MF global-shortcuts-provider/gnome-control-center-global-shortcuts-provider.p/cc-global-shortcut-dialog.c.o.d -o global-shortcuts-provider/gnome-control-center-global-shortcuts-provider.p/cc-global-shortcut-dialog.c.o -c ../global-shortcuts-provider/cc-global-shortcut-dialog.c
␛[01m␛[K../global-shortcuts-provider/cc-global-shortcut-dialog.c:38:10:␛[m␛[K ␛[01;31m␛[Kfatal error: ␛[m␛[Kpanels/keyboard/cc-keyboard-resources.h: No such file or directory
   38 | #include ␛[01;31m␛[K"panels/keyboard/cc-keyboard-resources.h"␛[m␛[K
      |          ␛[01;31m␛[K^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~␛[m␛[K
---
 panels/keyboard/meson.build | 5 ++++-
 1 file changed, 4 insertions(+), 1 deletion(-)

diff --git a/panels/keyboard/meson.build b/panels/keyboard/meson.build
index cd1b54e8e1..18a2f10dea 100644
--- a/panels/keyboard/meson.build
+++ b/panels/keyboard/meson.build
@@ -74,13 +74,15 @@ sources = files(
   'cc-input-source-xkb.c',
 )
 
-sources += gnome.compile_resources(
+keyboard_resources_sources = gnome.compile_resources(
   'cc-' + cappletname + '-resources',
   cappletname + '.gresource.xml',
   c_name: 'cc_' + cappletname,
   export: true
 )
 
+sources += keyboard_resources_sources
+
 deps = common_deps + [
   keyboard_shortcuts_dep,
   gnome_desktop_dep,
@@ -110,6 +112,7 @@ panels_libs += keyboard_panel_lib
 
 keyboard_panel_dep = declare_dependency(
   include_directories: [top_inc, keyboard_inc],
+  sources: [keyboard_resources_sources[1]],
   link_with: keyboard_panel_lib,
 )
 
-- 
GitLab


From fcfa9de7817c78c017d6f762043bdb6da5c378d7 Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Pablo=20Correa=20G=C3=B3mez?= <ablocorrea@hotmail.com>
Date: Wed, 28 May 2025 17:35:31 +0200
Subject: [PATCH 2/2] network: include generated resource header files in the
 panel dep

This should fix a rare race where the network tests panel has its
sources compiled before the network resources are generated. Per the
meson documentation sources in declare_dependency: "sources to add to
targets (or generated header files that should be built before sources
including them are built)". Without the sources, the dependency makes
sure that the global shortcuts will not be linked until the dependency
is ready, but some of its files might be compiled before the header is
ready.

Example error:
https://gitlab.alpinelinux.org/fossdd/aports/-/jobs/1872469

ninja: job failed: cc -Itests/network/test-wifi-panel-text.p -Itests/network -I../tests/network -I. -I.. -Ipanels/network -I../panels/network -I../tests/network/nm-utils -Ipanels/common -I../panels/common -Ishell -I/usr/include/glib-2.0 -I/usr/lib/glib-2.0/include -I/usr/include/libmount -I/usr/include/blkid -I/usr/include/gsettings-desktop-schemas -I/usr/include/libadwaita-1 -I/usr/include/gtk-4.0 -I/usr/include/pango-1.0 -I/usr/include/harfbuzz -I/usr/include/gdk-pixbuf-2.0 -I/usr/include/webp -I/usr/include/cairo -I/usr/include/freetype2 -I/usr/include/libpng16 -I/usr/include/pixman-1 -I/usr/include/graphene-1.0 -I/usr/lib/graphene-1.0/include -I/usr/include/fribidi -I/usr/include/appstream -I/usr/include/gio-unix-2.0 -I/usr/include/libnm -I/usr/include/libnma -I/usr/include/libmm-glib -I/usr/include/ModemManager -flto=auto -fdiagnostics-color=always -D_FILE_OFFSET_BITS=64 -Wall -Winvalid-pch -DHAVE_CONFIG_H -Wno-deprecated-declarations -Os -fstack-clash-protection -Wformat -Werror=format-security -g -fPIE -pthread '-DTEST_SRCDIR="/builds/fossdd/aports/community/gnome-control-center/src/gnome-control-center-48.2/tests/network"' -DNETWORKMANAGER_COMPILATION=NM_NETWORKMANAGER_COMPILATION_WITH_GLIB -DNETWORKMANAGER_COMPILATION_TEST '-DTEST_NM_SERVICE="/builds/fossdd/aports/community/gnome-control-center/src/gnome-control-center-48.2/tests/network/nm-utils/test-networkmanager-service.py"' -MD -MQ tests/network/test-wifi-panel-text.p/test-wifi-text.c.o -MF tests/network/test-wifi-panel-text.p/test-wifi-text.c.o.d -o tests/network/test-wifi-panel-text.p/test-wifi-text.c.o -c ../tests/network/test-wifi-text.c
In file included from ../tests/network/test-wifi-text.c:34:
../panels/network/cc-wifi-panel.c:20:10: fatal error: cc-network-resources.h: No such file or directory
   20 | #include "cc-network-resources.h"
      |          ^~~~~~~~~~~~~~~~~~~~~~~~
---
 panels/network/meson.build | 9 ++++++++-
 tests/network/meson.build  | 8 +++-----
 2 files changed, 11 insertions(+), 6 deletions(-)

diff --git a/panels/network/meson.build b/panels/network/meson.build
index aae15edb6b..e36fee3eff 100644
--- a/panels/network/meson.build
+++ b/panels/network/meson.build
@@ -47,12 +47,13 @@ sources = files(
   'ui-helpers.c'
 )
 
-sources += gnome.compile_resources(
+network_resources_sources = gnome.compile_resources(
   'cc-' + cappletname + '-resources',
   cappletname + '.gresource.xml',
   c_name: 'cc_' + cappletname,
   export: true
 )
+sources += network_resources_sources
 
 network_panel_lib = static_library(
   cappletname,
@@ -63,3 +64,9 @@ network_panel_lib = static_library(
   link_with: libconnection_editor
 )
 panels_libs += network_panel_lib
+
+network_panel_dep = declare_dependency(
+  include_directories: [top_inc, network_inc],
+  sources: [network_resources_sources[1]],
+  link_with: network_panel_lib,
+)
diff --git a/tests/network/meson.build b/tests/network/meson.build
index 1ace0445da..8912d68448 100644
--- a/tests/network/meson.build
+++ b/tests/network/meson.build
@@ -1,6 +1,6 @@
 
 
-includes = [top_inc, include_directories('../../panels/network', 'nm-utils')]
+includes = [top_inc, include_directories('nm-utils')]
 cflags = [
   '-DTEST_SRCDIR="@0@"'.format(meson.current_source_dir()),
   '-DNETWORKMANAGER_COMPILATION=NM_NETWORKMANAGER_COMPILATION_WITH_GLIB',
@@ -22,8 +22,7 @@ if Xvfb.found()
     'test-network-panel',
     ['test-network-panel.c', 'cc-test-window.c', 'nm-utils/nm-test-utils-impl.c'],
     include_directories : includes + [common_inc],
-           dependencies : common_deps + network_manager_deps + [libtestshell_dep],
-              link_with : [network_panel_lib],
+           dependencies : common_deps + network_manager_deps + [libtestshell_dep, network_panel_dep],
                  c_args : cflags
   )
 
@@ -39,8 +38,7 @@ exe = executable(
   'test-wifi-panel-text',
   ['test-wifi-text.c'],
   include_directories : includes + [common_inc],
-  dependencies : common_deps + network_manager_deps + [libtestshell_dep],
-  link_with : [network_panel_lib],
+  dependencies : common_deps + network_manager_deps + [libtestshell_dep, network_panel_dep],
   c_args : cflags,
 )
 
-- 
GitLab

